### PR TITLE
remove outdated build instructions

### PR DESCRIPTION
**Pull request by justincely**
Signed-off-by: Justin Ely <justincely@gmail.com>

Commit Message: remove outdated build instructions
Additional Description: Some instructions in the docs reference folders and scripts that don't exist.  Removing these instructions to prevent confusion. 
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Fixes #116790

### CODE REVIEW

**Comment by justincely** [comment]
closes #16790

**Review by phlax — APPROVED** [review]
lgtm - thanks @justincely 

cc @lizan  for confirmation that this is obsolete

**Comment by mattklein123** [comment]
I think these instructions are probably still generally relevant and should just refer to here? https://github.com/envoyproxy/envoy-build-tools

I think it's still possible to locally test changes to the build image? So perhaps either adjust to point to the other repo or move the instructions to the other repo and then point there?

**Comment by justincely** [comment]
So i'm definitely not familiar with the build-tools, but it doesn't seem to still be relevant.  A quick look through that repo shows the scripts have been changed quite a bit from these instructions.  I tried running a few scripts (kind randomely because the README's don't say much) and run into errors with a few.

Maybe it's enough to just tell users to go to `envoy-build-tools` if they need to modify the build image and let the maintainers of that repo handle fixing the instructions if people need it?

**Comment by phlax** [comment]
i got a bit lost trying to track what these instructions related to and gave up

@lizan probably knows

it seems to allude to creating your own build image, but doesnt suggest any changes.

Also not sure why you would want to do it this way - seems to me it would be a lot easier to create a new Dockerfile based on the upstream build one and to add any hackery there - but maybe im missing the point

**Comment by mattklein123** [comment]
The point of these instructions is to tell people how to test a build image change, but the build image no longer lives in this repo, it lives in the other repo. Instead of deleting this entirely I would at minimum link to the other repo and tell people to go there, make a new build image via Dockerfile hackery or whatever, and then use that for testing, etc.

**Comment by justincely** [comment]
I took a stab it it.  I don't know the instructions on the docker hackery, so just kept it generic.  I think if you want more specifics, someone with knowledge of that may need to update it.

File: ci/README.md [hunk 1]
```diff
@@ -172,20 +172,10 @@ From an interactive session, you can invoke `bazel` manually or use the `./ci/wi
...
-DISTRO=ubuntu
-cd ci/build_container
-LINUX_DISTRO="${DISTRO}" CIRCLE_SHA1=my_tag ./docker_build.sh  # Wait patiently for quite some time
-cd ../..
-IMAGE_NAME="envoyproxy/envoy-build-${DISTRO}" IMAGE_ID=my_tag ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.whatever'
-```
-
-This build the Ubuntu based `envoyproxy/envoy-build-ubuntu` image, and the final call will run against your local copy of the build image.
+The base build image used in the CI flows here lives in the [envoy-build-tools](https://github.com/envoyproxy/envoy-build-tools)
+repository. If you needto make and/or test changes to the build image, instructions to do so can be found in
```
Comment (mattklein123) [inline ci/README.md:176]:
```suggestion
repository. If you need to make and/or test changes to the build image, instructions to do so can be found in
```

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks LGTM with small typo.

/wait

**Comment by justincely** [comment]
thx, committed your suggestion

**Pull request merged**